io/include/detray/io/backend/detail/grid_reader.hpp

```diff
@@ -297,9 +344,35 @@ class grid_reader {
         using axes_t =
             axis::multi_axis<false, local_frame_t,
                              axis::single_axis<bounds_ts, binning_ts>...>;
+
         using bin_t =
             std::conditional_t<bin_capacity == 0, bins::dynamic_array<value_t>,
                                bins::static_array<value_t, bin_capacity>>;
+        auto print_bounds = [&] {
+            std::stringstream os;
+            std::size_t i = 0;
+            auto helper = [&os, &i]<typename T>(T /*arg*/) {
+                if (i > 0) {
+                    os << ", ";
+                }
+                i++;
+                os << T::type << "<" << T::label << ">";
+            };
+
+            (helper(bounds_ts{}), ...);
+            return os.str();
+        };
+
+        DETRAY_DEBUG("Now building grid. Recap:");
```

Comment:
Why not just print the single axes and the frame and bin types? The binnings and bounds should be contained and it is still more readable than printing the whole `grid_t`

Reply:
Can change of course but I was going got maximum verbosity